datahub-web-react/src/providers/EducationStepsProvider.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useEffect, useState } from 'react';
+import { useGetAuthenticatedUser } from '../app/useGetAuthenticatedUser';
+import { getStepIds } from '../app/onboarding/utils';
+import { useBatchGetStepStatesQuery } from '../graphql/step.generated';
+import { EducationStepsContext } from './EducationStepsContext';
+import { StepStateResult } from '../types.generated';
+
+export function EducationStepsProvider({ children }: { children: React.ReactNode }) {
+    const userUrn = useGetAuthenticatedUser()?.corpUser.urn;
+    const stepIds = getStepIds(userUrn || '');
+    const { data } = useBatchGetStepStatesQuery({ variables: { input: { ids: stepIds } } });
+    const results = data?.batchGetStepStates.results;
+    const [educationSteps, setEducationSteps] = useState<StepStateResult[] | null>(null);
+
+    useEffect(() => {
+        if (results && !educationSteps?.length) {
+            setEducationSteps(results);
+        }
```

Comment:
if we start from scratch and get an empty list for `results`, will this put us into an infinite loop? maybe our dependency arrow below is preventing that

Reply:
but it still might be safer to say `if (results && educationSteps === null)` since I think this is jut meant to populate `educationSteps` on initial load

Reply:
Good suggestion, refactored